### PR TITLE
Fix flaky test

### LINKED ISSUE #1922: tests.integration.service_test test_scale_with_custom_container_name_outputs_warning
(Aliased as "Test that calling scale on a service that has a custom container name" from the docstring)

Seems to be failing fairly frequently against docker 1.8.1

* https://jenkins.dockerproject.org/job/Compose-PRs/1118/console
* https://jenkins.dockerproject.org/job/Compose-PRs/1122/console
* https://jenkins.dockerproject.org/job/Compose-PRs/1117/console
* https://jenkins.dockerproject.org/job/Compose-PRs/1115/console
* https://jenkins.dockerproject.org/job/Compose-PRs/1112/console

(a sample from today)

I suspect there is an underlying docker bug here. We should confirm and make sure it's known upstream, and/or find a way to make the test suite less flaky.

### PR DESCRIPTION
**Pull request by dnephin**
Branched from #1939
Resolves #1922 

This works around the problem of containers not being removed correctly by using a different name for the service.  Also adds a new label to all containers so that if we end up with test pollution we can track the container back to the test that created it.

This is definitely just a workaround, but I think the root issue is actually a race condition in the docker engine so will take much longer to fix.  I don't like having every branch fail our test suite. So I'd like to push this workaround, since the failure isn't directly related to compose itself.

### CODE REVIEW

**Comment by dnephin** [comment]
https://jenkins.dockerproject.org/job/Compose-PRs/1179/console

**Comment by dnephin** [comment]
It worked! build is green

File: tests/integration/state_test.py [hunk 1]
```diff
@@ -199,7 +199,6 @@ def test_trigger_start(self):
 
         self.assertEqual([c.is_running for c in containers], [False, True])
 
-        web = self.create_service('web', **options)
```
Comment (aanand) [inline tests/integration/state_test.py]: What's the purpose of this change?

Comment (dnephin) [inline tests/integration/state_test.py]: I found it while debugging. It's a no-op since line 193 does the same thing.

**Comment by aanand** [comment]
LGTM
